README.md

```diff
@@ -42,10 +42,13 @@ for examples of this and other features.
 
 ## Patterns
 
-* `_` matches anything
-* `x` (an identifier) matches anything, binds value to the variable `x`
+* `_` matches any value
+* `x` (an identifier) matches any value and binds it to the variable `x`
 * `T(x,y,z)` matches structs of type `T` with fields matching patterns `x,y,z`
-* `T(y=1)` matches structs of type `T` whose `y` field equals `1`
+* `T(y=p)` matches structs of type `T` whose `y` field matches pattern `p`
+* `(;x,y,z)` matches values with fields `x,y,z` binding to variables `x,y,z`
+* `(;x=p)` matches values with field `x` matching pattern `p`; also binds the field to `x`
```

Comment:
(I know this is not open for review, but...)

I don't think binding the name `x` is a good idea. For example, consider a binary tree. What if I want to match

`Node(value=2, left=Node(value=1), right=Node(value=3))`

This can never match because `1` is not equal to `2` and `3`.

This kind of code arises in practice, for example when writing the code to balance a binary tree. These semantics would force the programmer to use positional notation.

Reply:
Good point. I had doubts about binding the field name as well.